Formula/groestlcoin.rb

```diff
@@ -0,0 +1,65 @@
+# typed: false
+# frozen_string_literal: true
+
+# Decentralized, peer to peer payment network
+class Groestlcoin < Formula
+  desc "Decentralized, peer to peer payment network"
+  homepage "https://groestlcoin.org/groestlcoin-core-wallet/"
+  url "https://github.com/Groestlcoin/groestlcoin/releases/download/v22.0/groestlcoin-22.0.tar.gz"
+  sha256 "6c412361e7ff84e9f7cf6a25be0d18742098ddc0f91688b368b0be3e6daa46f1"
+  license "MIT"
+  head "https://github.com/groestlcoin/groestlcoin.git", branch: "master"
+
+  livecheck do
+    url :url
+    strategy :github_latest
+  end
+
+  bottle do
+    sha256 cellar: :any,                 arm64_monterey: "b0704a6dc3a96f5bbb71d93a468b1a4881e747e90ddfa14d6ff0d97d5a23c6eb"
+    sha256 cellar: :any,                 arm64_big_sur:  "6c8caf0509daf986cd3af45eeea78789ccd4b0a8eb5aa1dfddaa624425e5b959"
+    sha256 cellar: :any,                 monterey:       "2859a13d7a48dad1a6701476d66ddb0b6940dcfe4b809b1c54512d78de247e26"
+    sha256 cellar: :any,                 big_sur:        "e8a1f3389faae4245c5f6a6ec8f179cb46e19af31f4e4637c6d3f24d55804b45"
+    sha256 cellar: :any,                 catalina:       "3279bd808f42d6ea608d04860ac276afda7c6ac2cf5d708469d4dc66672f210a"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "cb60fb85781435dfb545cea0c7e08339a7b3e01d60df604ccf9cba830f148e36"
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "berkeley-db"
+  depends_on "boost@1.76"
+  depends_on "libevent"
+  depends_on macos: :catalina
+  depends_on "miniupnpc"
+  depends_on "zeromq"
+
+  on_linux do
+    depends_on "util-linux" => :build # for `hexdump`
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5"
+
+  def install
+    ENV.delete("SDKROOT") if MacOS.version == :el_capitan && MacOS::Xcode.version >= "8.0"
+
+    system "./autogen.sh"
+    system "./configure", *std_configure_args,
+           "--disable-dependency-tracking",
+           "--disable-silent-rules",
+           "--with-incompatible-bdb",
+           "--with-boost-libdir=#{Formula["boost@1.76"].opt_lib}"
+    system "make", "install"
+    pkgshare.install "share/rpcauth"
+  end
+
+  service do
+    run opt_bin/"groestlcoind"
+  end
+
+  test do
+    system "#{bin}/groestlcoin-cli", "-version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.